attestation-agent/attester/Cargo.toml

```diff
@@ -44,7 +45,8 @@ all-attesters = ["tdx-attester", "sgx-attester", "az-snp-vtpm-attester", "az-tdx
 # quotes. It's an unconditional dependency for tdx-attester since that is the only way to
 # generate TDX quotes with upstream kernels.
 tsm-report = ["tempfile"]
-tdx-attester = ["tdx-attest-rs"]
+tdx-attester = ["tsm-report", "tdx-getquote-ioctl"]
```

Comment:
Some of this is explained in the comment: with this PR we make the TSM as the primary mechanism without the option to disable it. `tdx-getquote-ioctl` is used during the transition to until we get 6.7 updated but them I'm looking to disable it in our default builds. However, there might still be users who need the out-of-tree patches so they can enabled it in their builds.

The reason for the separate feature is that when disabled, we also get to drop `tdx-attest-rs` from the builds which is a maintenance nightmare: see #352

Reply:
I am all in favor of getting rid of the dcap dependency, but I would have just made those changes directly in the source when they were feasible rather than introducing a feature for it.

Reply:
I thought about it too but this gives the flexibility for CoCo upstream to move faster to configfs-tsm based setups while some downstreams that are using the TDX "out-of-tree" work can keep their deprecated ioctl getquote working.

Reply:
But for the record: with the addition of `check_init_data` things changed a bit since GET_REPORT *is* supported by the upstream ioctl. I'm also OK to drop that additional feature for now and add it later once the upstream story for the ioctl future is clear.

Reply:
dropped all the `cfg-if` for now